### ISSUE TITLE
Unspecified boolean(x) function present in annotations inside StateGraph library.

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#450]
**Reported by adrpo on 2 Nov 2010 15:27 UTC**
Hi all,

In the latest MSL 3.2, StateGraph.mo we have this annotation:
```
Placement(visible=not boolean(hideConnector), transformation(extent={{-140,-20},{-100,20}}, rotation=0)));
```
The function boolean(...) used in this annotation is not present in the specification. I assume it converts from something to boolean, but it should be present in the specification!

Cheers,
Adrian Pop/

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/450

**Comment by modelica-trac-importer** [comment]
**Comment by stefanv on 2 Nov 2010 15:33 UTC**
If there is such a currently unspecified function, it should probably be called Boolean, to correspond with the existing Integer() and String() functions.

**Comment by modelica-trac-importer** [comment]
**Comment by adrpo on 2 Nov 2010 15:36 UTC**
I think you're making a mistake. Integer(...) only takes Enumeration and integer(...) only takes Real. Don't ask me why we have two different Integer/integer for different types.

**Comment by modelica-trac-importer** [comment]
**Comment by stefanv on 2 Nov 2010 15:41 UTC**
I think Integer() is a cast (which doesn't actually do anything at run time) whereas integer() is a conversion. I guess whether [Bb]oolean() is a cast or a conversion would be implementation dependent.

**Comment by modelica-trac-importer** [comment]
**Comment by adrpo on 2 Nov 2010 15:44 UTC**
Integer(...) function only takes an Enumeration type. See Modelica Specification 3.2, Section 3.7.1 Numeric Functions and Conversion Functions, page 20. So Integer(...) is not a cast!

**Comment by modelica-trac-importer** [comment]
**Comment by HansOlsson on 2 Nov 2010 15:45 UTC**
I would assume it is just an error and should be removed.

**Comment by modelica-trac-importer** [comment]
**Comment by adrpo on 2 Nov 2010 16:01 UTC**
I think that boolean(hideConnector) can be safely replaced with hideConnector as hideConnector is Boolean anyway, so the operation is a no-op:
```
    parameter Boolean hideConnector=false
        "= true, if connector is not shown in the dynamic object diagram";
```

**Comment by modelica-trac-importer** [comment]
**Comment by hansolsson on 1 Sep 2011 15:18 UTC**
Language group:

Remove from library.

**Comment by modelica-trac-importer** [comment]
**Modified by hansolsson on 1 Sep 2011 15:18 UTC**

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 7 Mar 2012 15:29 UTC**
Fixed in 6c4fd286c37e3766c8c89a386c8031071755356e.
